apps/pkeyutl.c

```diff
@@ -671,15 +669,12 @@ static int setup_peer(EVP_PKEY_CTX *ctx, int peerform, const char *file,
     peer = load_pubkey(file, peerform, 0, NULL, engine, "peer key");
     if (peer == NULL) {
         BIO_printf(bio_err, "Error reading peer key %s\n", file);
-        ERR_print_errors(bio_err);
         return 0;
     }
 
-    ret = EVP_PKEY_derive_set_peer(ctx, peer);
+    ret = EVP_PKEY_derive_set_peer(ctx, peer) > 0;
 
     EVP_PKEY_free(peer);
-    if (ret <= 0)
-        ERR_print_errors(bio_err);
```

Comment:
Isn't this one lost?

Reply:
No - see how the function is used and what happens on error.

Reply:
BTW, there was a bug in this `setup_peer()` function: 
`EVP_PKEY_derive_set_peer(ctx, peer) < 0` was not recognized as error.